### ISSUE TITLE
KnpMenu sidebar current item for non-admin items

### ISSUE THREAD

**Issue by gcaseres** [sonata-project/SonataAdminBundle#3110]
I was trying to create some menu items with specific routes for the sonata sidebar in my config.yml like this:
```
news.admin.group.security:
    label: Security
        items:                    
            - route: fos_user_profile_edit_authentication
              label: Editar profile
            - route: fos_user_change_password
              label: Change password
```

It worked, but then I noticed than when I click on an item, it doesn't sets it as the current item.
So the sidebar menu keeps collapsed as if no item were active.

I tried to fix this, but I had to overwrite the sonata admin template for knp menu rendering.
I added (like in knp menu core implementation) the use of the matcher to detect if the item is currently selected:

```
{% block item %}
    {%- if item.displayed %}
        {#- check role of the group #}
        {%- set display = (item.extra('roles') is empty or is_granted('ROLE_SUPER_ADMIN') ) %}
        {%- for role in item.extra('roles') if not display %}
            {%- set display = is_granted(role) %}
        {%- endfor %}
    {%- endif %}


    {%- if item.displayed and display|default %}
        {%- set active = false %}
        {%- if item.extra('active') is not empty and item.extra('active') %}
            {%- set active = true %}
        {%- elseif item.extra('admin') is not empty and item.extra('admin').hasroute('list') and item.extra('admin').isGranted('LIST') and request.get('_sonata_admin') == item.extra('admin').code %}
            {%- set active = true %}
        {%- elseif item.route is defined and request.get('_route') == item.route %}
            {%- set active = true %}
        {%- elseif matcher.isCurrent(item) %}
            {%- set active = true %}
        {%- else %}
            {%- for child in item.children if not active %}
                {%- if child.extra('admin') is not empty and child.extra('admin').hasroute('list') and child.extra('admin').isGranted('LIST') and request.get('_sonata_admin') == child.extra('admin').code %}
                    {%- set active = true %}
                {%-  elseif child.route is defined and request.get('_route') == child.route %}
                    {%- set active = true %}
                {%- elseif matcher.isCurrent(child) %}
                    {%- set active = true %}
                {%- endif %}
            {%- endfor %}
        {%- endif %}

        {%- if item.hasChildren %}
            {%- do item.setAttribute('class', (item.attribute('class')~' treeview')|trim) %}
        {%- endif %}
        {%- if active %}
            {%- do item.setAttribute('class', (item.attribute('class')~' active')|trim) %}
            {%- do item.setChildrenAttribute('class', (item.childrenAttribute('class')~' active')|trim) %}
        {%- endif %}

        {%- do item.setChildrenAttribute('class', (item.childrenAttribute('class')~' treeview-menu')|trim) %}
        {{ parent() }}
    {% endif %}
{% endblock %}

```

Shouldn't this modification be applied to the sonata project?

The specific lines added are:

For item:

```
                {%- elseif matcher.isCurrent(item) %}
                    {%- set active = true %}
```

For child:
```
                {%- elseif matcher.isCurrent(child) %}
                    {%- set active = true %}
```

Sorry for my bad english.

**Comment by hason** [comment]
This issue is fixed in PR #3065.

**Comment by greg0ire** [comment]
Please reopen if this still isn't fixed in Sonata 3.
